Assets/Scripts/Layers/LayerTypes/GeoJSONPolygonLayer.cs

```diff
@@ -49,44 +52,50 @@ public void AddAndVisualizeFeature<T>(Feature feature, CoordinateSystem original
             if (SpawnedVisualisations.Any(f => f.feature.GetHashCode() == feature.GetHashCode()))
                 return;
 
-            // Create visual with random color if enabled
-            Material featureRenderMaterial = GetMaterialInstance();
-
-            // Add visualisation to the layer, and store it in the SpawnedVisualisations list where we tie our Feature to the visualisations
             var newFeatureVisualisation = new FeaturePolygonVisualisations { 
                 feature = feature,
                 geoJsonPolygonLayer = this
             };
-            if (feature.Geometry is MultiPolygon multiPolygon)
-            {
-                var polygonVisualisations = GeoJSONGeometryVisualizerUtility.VisualizeMultiPolygon(multiPolygon, originalCoordinateSystem, featureRenderMaterial);
-                newFeatureVisualisation.AppendVisualisations(polygonVisualisations);
-            }
-            else if (feature.Geometry is Polygon polygon)
+            Material featureRenderMaterial = GetMaterialInstance();
+
+            // Add visualisation to the layer, and store it in the SpawnedVisualisations list where we tie our Feature
+            // to the visualisations
+            switch (feature.Geometry)
             {
-                var singlePolygonVisualisation = GeoJSONGeometryVisualizerUtility.VisualizePolygon(polygon, originalCoordinateSystem, featureRenderMaterial);
-                newFeatureVisualisation.AppendVisualisations(singlePolygonVisualisation);
+                case MultiPolygon multiPolygon:
+                    newFeatureVisualisation.AppendVisualisations(GeometryVisualizationFactory.CreatePolygonVisualization(
+                        multiPolygon, 
+                        originalCoordinateSystem, 
+                        featureRenderMaterial
+                    ));
+                    break;
+                case Polygon polygon:
+                    newFeatureVisualisation.AppendVisualisations(GeometryVisualizationFactory.CreatePolygonVisualisation(
+                        polygon, 
+                        originalCoordinateSystem, 
+                        featureRenderMaterial
+                    ));
+                    break;
             }
 
             SpawnedVisualisations.Add(newFeatureVisualisation);
         }
 
+        public void ApplyStyling()
+        {
+            SpawnedVisualisations.ForEach(visualisations => visualisations.SetMaterial(GetMaterialInstance()));
```

Comment:
waarom een linq query hier gebruiken ipv een normale foreach loop (dat vind ik leesbaarder)?

Reply:
Kan ik doen :) Ik vind soms simpele foreaches prettiger om in een Linq statement te doen omdat het voor mij 1 ding is en ("Apply all" idee), versus allemaal losse handelingen. Maar: het heeft geen reden anders dan dat ik dat prettig vond :D

Reply:
Ik commit het zometeen - maar ik heb het omgezet naar een klassieke foreach; wel met braces, want dat moet ;)